### ISSUE TITLE
Redesign of the logging architecture

### ISSUE THREAD

**Issue by cvandeplas** [MISP/MISP#1450]
The logging architecture has many bad designs and limitations and should be completely re-written.
For example it is logging changes in the data model in a very generic way.

Rewriting this is a considerable effort.
This issue is made for long-term tracking and can be used to propose and document better logging-alternatives. 

Related also #1291

**Comment by adulau** [comment]
Indeed. Some notes, the logging should be separated from the current data-storage:
- To allow large MISP instance to logs everything as they wish without impacting the MISP usage. (e.g. API logging is clearly something needed for audit purposes)
- To allow the use of specific log entries (especially tuple like org,api key/user) to notify ORG admin about specific issues. (e.g. which API is revoked but still used, which API Key is misconfigured or which API key is used from which IP)

The idea was to keep full entry logs into a simple syslog-like format but at the same time, the logs are updating the records that are used for the API notification problems or log search.

Maybe the best would be to have standalone process having a SUB channel where all the logs done by MISP are published.
